### PR TITLE
ci: add IPv6 support to `NetworkManager`

### PR DESCRIPTION
**Pull request by thevar1able**
### Changelog category (leave one):
- CI Fix or Improvement (changelog entry is not required)

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [x]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit c65e73d1a582f9b344bfda66f6efa1803883030a with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/68900/c65e73d1a582f9b344bfda66f6efa1803883030a/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Push to Dockerhub</td><td>The check for building and pushing the CI related docker images to docker hub</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/68900/c65e73d1a582f9b344bfda66f6efa1803883030a/push_to_dockerhub_aarch64.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Comment by thevar1able** [comment]
`test_replicated_database_cluster_groups/test.py::test_cluster_groups` https://github.com/ClickHouse/ClickHouse/issues/69038

File: tests/integration/helpers/network.py [hunk 1]
```diff
@@ -178,6 +279,20 @@ def clean_all_user_iptables_rules():
                     + " iterations, last error: "
                     + str(res.stderr)
                 )
+                break
```
Comment (Felixoid) [inline tests/integration/helpers/network.py:282]: maybe, better to do `for iptabes in ('iptables', 'ip6tables')`?

File: tests/integration/helpers/network.py [hunk 2]
```diff
@@ -59,16 +111,36 @@ def create_rule(src, dst):
...
         while self._iptables_rules:
             rule = self._iptables_rules.pop()
-            _NetworkManager.get().delete_iptables_rule(**rule)
+
+            if self._is_ipv6_rule(rule):
+                _NetworkManager.get().delete_ip6tables_rule(**rule)
+            else:
+                _NetworkManager.get().delete_iptables_rule(**rule)
+            # _NetworkManager.get().delete_iptables_rule(**rule)
+            # _NetworkManager.get().delete_ip6tables_rule(**rule)
```
Comment (Felixoid) [inline tests/integration/helpers/network.py:143]: leftovers?

File: tests/integration/helpers/network.py [hunk 3]
```diff
@@ -59,16 +111,36 @@ def create_rule(src, dst):
...
+                "action": action,
+            }
+            if port is not None:
+                rule["destination_port"] = port
+            return rule
+
         self._add_rule(create_rule(left, right))
         self._add_rule(create_rule(right, left))
 
+        if left.ipv6_address and right.ipv6_address:
```
Comment (Felixoid) [inline tests/integration/helpers/network.py:127]: The upper lines are without conditions. Does it mean we always want to create ipv4 unconditionally?

**Review by Felixoid — COMMENTED** [review]
I'd like to see it work in the CI before we merge it.

File: tests/integration/helpers/network.py (hunk 3 above)
Comment (thevar1able) [reply tests/integration/helpers/network.py:127]: Right.

**Review by evillique — APPROVED** [review]

**Pull request merged**